scripts/src/java/org/oppia/android/scripts/coverage/CoverageRunner.kt

```diff
@@ -0,0 +1,79 @@
+package org.oppia.android.scripts.coverage
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import org.oppia.android.scripts.common.BazelClient
+import org.oppia.android.scripts.common.CommandExecutor
+import org.oppia.android.scripts.common.CommandExecutorImpl
+import org.oppia.android.scripts.common.ScriptBackgroundCoroutineDispatcher
+import java.io.File
+import java.util.concurrent.TimeUnit
+
+/**
+ * Class responsible for running coverage analysis asynchronously.
+ *
+ * @param repoRoot the absolute path to the working root directory
+ * @param targetFile Path to the target file to analyze coverage.
+ */
+class CoverageRunner(
+  private val repoRoot: File,
+  private val scriptBgDispatcher: ScriptBackgroundCoroutineDispatcher
+) {
+
+  /**
+   * Runs coverage analysis asynchronously for the Bazel test target.
+   *
+   * @param repoRoot the absolute path to the working root directory
+   * @param scriptBgDispatcher the [ScriptBackgroundCoroutineDispatcher] to be used for running the coverage command
+   * @param bazelTestTarget Bazel test target to analyze coverage.
+   * @return a deferred value that contains the path of the coverage data file [will contain the proto for the coverage data].
+   */
+  suspend fun runWithCoverageAsync(
+    bazelTestTarget: String
+  ): Deferred<String?> {
+    return CoroutineScope(scriptBgDispatcher).async {
+      val coverageData = getCoverage(bazelTestTarget)
+      val data = coverageData
+      parseCoverageDataFile(data)
+    }
+  }
+
+  /**
+   * Runs coverage command for the Bazel test target.
+   *
+   * @param repoRoot the absolute path to the working root directory
+   * @param scriptBgDispatcher the [ScriptBackgroundCoroutineDispatcher] to be used for running the coverage command
+   * @param bazelTestTarget Bazel test target to analyze coverage.
+   * @return a lisf of string that contains the result of the coverage execution.
+   */
+  fun getCoverage(
+    bazelTestTarget: String
+  ): List<String> {
+    val commandExecutor: CommandExecutor = CommandExecutorImpl(
```

Comment:
I suggest passing this in like we do for other utilities. You can default its construction, but being able to configure it can be really useful in the future if we need to tweak the runtime (in case some code coverage runs take longer than 5 minutes to complete).

Reply:
I agree. Implementing `processTimeout` via command-line arguments would be valuable, especially for managing longer processes like coverage runs, as I too have seen many failing due to this and have had to rerun them using cached results. 

**Custom Process Timeout:** [[commit](https://github.com/oppia/oppia-android/pull/5423/commits/5e6d2ef8288011ac57c8d9ac1a1859f66474a305)]

I've updated it to accept `processTimeout` as an argument, allowing **dynamic timing** or **defaulting to 5 minutes**. I would like to know your thoughts on this approach. If it adds complexity at this stage, I can revert it back to the default of 5 minutes for all processes.

Reply:
I think your approach works. I wasn't thinking about making the script itself configurable, just the script's logic class. However, if you think downstream code will need to customize the execution timeout then this approach makes sense.